### PR TITLE
[SUPERMASSIVE] added subscribe operation

### PR DESCRIPTION
**Pull request by vejrj**

### CODE REVIEW

File: packages/supermassive/src/__tests__/subscribe.test.ts [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+    },
+  },
+];
+
+describe("subscribeWithSchema", () => {
+  testCases.forEach(({ name, document, variables }: TestCase) => {
+    it(name, async () => {
+      await compareResultsForSubscribeWithSchema(document, variables);
+    });
+  });
```
Comment (vladar) [inline packages/supermassive/src/__tests__/subscribe.test.ts:57]: Can we switch to [`test.each` or `it.each` syntax](https://jestjs.io/docs/api#testeachtablename-fn-timeout) - IDEs support it a bit better than custom loops.

File: change/@graphitation-supermassive-51bd644d-9254-42bb-a0c4-c6653f90b7d1.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "type": "minor",
+  "comment": {
```
Comment (freiksenet) [inline change/@graphitation-supermassive-51bd644d-9254-42bb-a0c4-c6653f90b7d1.json:3]: This should be a string comment describing change.

File: examples/supermassive-todomvc/src/schema/link.ts [hunk 3]
```diff
@@ -1,10 +1,17 @@
-import { execute as graphQLExecute, isInputType, parse } from "graphql";
+import {
```
Comment (freiksenet) [inline examples/supermassive-todomvc/src/schema/link.ts:1]: [nit] unused imports

File: packages/supermassive/src/subscribeWithoutSchema.ts [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+
+import {
+  ObjectTypeResolver,
+  FunctionFieldResolver,
+  FieldResolver,
+  Resolvers,
+  ExecutionResult,
+} from "./types";
+
+export interface SubscriptionArgs {
```
Comment (freiksenet) [inline packages/supermassive/src/subscribeWithoutSchema.ts:36]: Are those different from execute args? Seems they are the same in graphql-js, let's unify it here too.

File: packages/supermassive/src/subscribeWithoutSchema.ts [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+      name: {
+        kind: Kind.NAME,
+        value: "String",
+      },
+    };
+  } else {
+    returnTypeNode = fieldNodes[0].__type as TypeNode;
+    returnTypeName = typeNameFromAST(returnTypeNode);
+    const typeResolvers = exeContext.resolvers[typeName];
+    resolveFn = ((typeResolvers as
```
Comment (freiksenet) [inline packages/supermassive/src/subscribeWithoutSchema.ts:238]: Are those any casts still needed? I think the type should support this now.

**Review by freiksenet — COMMENTED** [review]
Looking good, some really minor issues. Great work!

File: packages/supermassive/src/subscribeWithoutSchema.ts (hunk 4 above)
Comment (vejrj) [reply packages/supermassive/src/subscribeWithoutSchema.ts:36]: We added `subscribeFieldResolver`

File: change/@graphitation-supermassive-51bd644d-9254-42bb-a0c4-c6653f90b7d1.json [hunk 6]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
+  "comment": "[SUPERMASSIVE] added subscribe operation",
```
Comment (freiksenet) [inline change/@graphitation-supermassive-51bd644d-9254-42bb-a0c4-c6653f90b7d1.json:3]:
```suggestion
  "comment": "Added support for subscribe operation",
```

File: packages/supermassive/src/subscribeWithoutSchema.ts (hunk 4 above)
Comment (freiksenet) [reply packages/supermassive/src/subscribeWithoutSchema.ts:36]: You can add it to all ExecutionArgs, like in graphql-js.

https://github.com/graphql/graphql-js/blob/main/src/execution/execute.ts#L154

**Review by freiksenet — APPROVED** [review]

**Pull request merged**